src/wallet/wallet.cpp

```diff
@@ -2661,6 +2662,13 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
 
         if (recipient.fSubtractFeeFromAmount)
             nSubtractFeeFromAmount++;
+
+        // Check if any destination contains a witness program:
+        int witnessversion = 0;
+        std::vector<unsigned char> witnessprogram;
+        if (recipient.scriptPubKey.IsWitnessProgram(witnessversion, witnessprogram)) {
```

Comment:
Should probably check/track witness version here? It shouldn't be >0 unless we update, but probably more future-safe.

Reply:
Would new witness versions require different change address behavior?

Reply:
Let's say you send to a v1, you wouldn't want to accidentally make a v0 to "copy" it. Perhaps more appropriate for an assert here.

Reply:
It might be assertophobia, but I'm not sure what you mean by that example. If you send coins to a future v1 segwit address, the change has to go somewhere right? So the change would just go to a v0 witness derived from the next available key in the pool.

Note that `IsWitnessProgram` writes to `witneessversion`, not the other way around.

It depends on what the intention of v1 is whether change can safely go to a v1 address. But I don't see how it would be unsafe to send to a v0 address. Of course maybe there is a reason, or future us will find a reason. So an assert would force them to think about that. Is that what you mean?

Reply:
Not unsafe, but defeats the privacy logic that this enables, I think.

Maybe I'm making too much hay of this.

Reply:
Really not sure what the point is here - we definitely should allowe sending to >v0 witness versions, its not our job to make sure someone giving us an address gives us something that is currently safe (ie has been forked to have some meaning), and we shouldn't need an update just to send to a newly-defined segwit version.

Reply:
@TheBlueMatt I don't care about that, I care that you are losing privacy if someone decides to.(though as noted elsewhere that would require a wallet update, so they just have to be updated in lock-step)

Reply:
@instagibbs  since we don't support v1  there is no 'private' option possible for us,  so given that privacy is impossible, we should at least choose the most efficient!

Reply:
This can be avoided/skipped if there is a custom address in coin control. Move it below to where `any_destination_native_segwit` is needed.